hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusDeleteTopicHandler.java

```diff
@@ -70,6 +74,18 @@ public void handle(@NonNull final HandleContext context) {
         requireNonNull(context, "The argument 'context' must not be null");
 
         final var op = context.body().consensusDeleteTopicOrThrow();
+
+        final var fees = context.feeCalculator(SubType.DEFAULT).legacyCalculate(sigValueObj -> {
+            try {
+                final var protoBody = fromPbj(context.body());
+                return ConsensusServiceFeeBuilder.getConsensusDeleteTopicFee(protoBody, sigValueObj);
+            } catch (InvalidTxBodyException e) {
+                throw new RuntimeException(e);
+            }
+        });
+
+        context.feeAccumulator().charge(context.payer(), fees);
```

Comment:
Will it work if we have this as API in `TransactionHandler` interface and if it can be called from `HandleWorkflow`, before calling `handle()` ?

Reply:
I agree, that would be preferable. We will need it when we fix the ingest workflow. For now I am just getting these in to get those tests passed, and then I will go through and move to another method. There were some complexities to adding the other method that was slowing me down, and I decided to just get in what I could and learn more about the requirements and then move these into specific methods later.